core/templates/pages/exploration-player-page/layout-directives/audio-bar.component.spec.ts

```diff
@@ -0,0 +1,568 @@
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AudioBarComponent);
+    component = fixture.componentInstance;
+    audioPlayerService = TestBed.inject(AudioPlayerService);
+    audioBarStatusService = TestBed.inject(AudioBarStatusService);
+    audioTranslationLanguageService = TestBed.inject(
+      AudioTranslationLanguageService);
+    audioPreloaderService = TestBed.inject(AudioPreloaderService);
+    assetsBackendApiService = TestBed.inject(AssetsBackendApiService);
+    audioTranslationManagerService = TestBed.inject(
+      AudioTranslationManagerService);
+    autogeneratedAudioPlayerService = TestBed.inject(
+      AutogeneratedAudioPlayerService);
+    playerPositionService = TestBed.inject(PlayerPositionService);
+
+    fixture.detectChanges();
+  });
+  beforeEach(() => {
+    spyOn(audioBarStatusService, 'markAudioBarExpanded').and.returnValue(null);
+    spyOn(audioBarStatusService, 'markAudioBarCollapsed').and.returnValue(null);
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
+  });
+
+  it('should set secondary audio translations when audio bar ' +
+    'is opened and audio is playing', fakeAsync(() => {
+    let params = {
+      audioTranslations: {},
+      componentName: 'feedback',
+      html: ''
+    };
+    let mockOnAutoplayAudioEventEmitter = new EventEmitter();
+    spyOnProperty(audioPlayerService, 'onAutoplayAudio')
+      .and.returnValue(mockOnAutoplayAudioEventEmitter);
+    let secondaryTranslaionsSpy = spyOn(
+      audioTranslationManagerService, 'setSecondaryAudioTranslations')
+      .and.returnValue(null);
+
+    component.ngOnInit();
+    component.expandAudioBar();
+    component.isPaused = false;
+    fixture.detectChanges();
+
+    mockOnAutoplayAudioEventEmitter.emit(params);
+    flush();
+    discardPeriodicTasks();
+    fixture.detectChanges();
+
+
+    expect(secondaryTranslaionsSpy).toHaveBeenCalledWith(
+      params.audioTranslations, params.html, params.componentName);
+  }));
+
+  it('should set current time when calling \'setProgress\'', () => {
+    // This time period is used to set progress
+    // when user pull the drag button in audio bar.
+    let param = {
+      value: 100
+    };
+    let currentTimeSpy = spyOn(
+      audioPlayerService, 'setCurrentTime').and.returnValue(null);
+
+    component.setProgress(param);
+
+    expect(currentTimeSpy).toHaveBeenCalledWith(100);
+  });
+
+  it('should check whether the auto generated language ' +
+    'code is selected', () => {
+    let autoGeneratedLanguageSpy = spyOn(
+      audioTranslationLanguageService, 'isAutogeneratedLanguageCodeSelected')
+      .and.returnValue(false);
+    let result = component.isAutogeneratedLanguageCodeSelected();
+
+    expect(result).toBe(false);
+    expect(autoGeneratedLanguageSpy).toHaveBeenCalled();
+  });
+
+  it('should check if the audio bar is available', () => {
+    // Audio bar is only accessible if the available
+    // languages are greater than one.
```

Comment:
nit: `...if the number of available languages is greater than one`.

Reply:
done